src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs

```diff
@@ -3392,5 +3363,85 @@ private void EmitBox(TypeSymbol type, SyntaxNode syntaxNode)
             _builder.EmitOpCode(ILOpCode.Box);
             EmitSymbolToken(type, syntaxNode);
         }
+
+        private void EmitCalli(BoundFunctionPointerInvocation ptrInvocation, UseKind useKind)
+        {
+            FunctionPointerMethodSymbol method = ptrInvocation.FunctionPointer.Signature;
+
+            EmitExpression(ptrInvocation.InvokedExpression, used: true);
+            LocalDefinition temp = null;
+            if (ptrInvocation.Arguments.Length > 0)
+            {
+                temp = AllocateTemp(ptrInvocation.InvokedExpression.Type, ptrInvocation.Syntax);
```

Comment:
It's potentially side-effecting. The token for the function pointer must be on top of the stack, but as some of my tests show this can be anything, such as a function call. We need to preserve left-to-right semantics of the code, however, so we need to save it to a temp.

Reply:
Cool, can you leave a comment about that? It wasn't obvious how that was side effecting. An example would be helpful